src/OpenTelemetry/Metrics/MetricStreamIdentity.cs

```diff
@@ -101,6 +104,8 @@ public MetricStreamIdentity(Instrument instrument, MetricStreamConfiguration? me
 
     public string MeterVersion { get; }
 
+    public IEnumerable<KeyValuePair<string, object?>> MeterTags { get; }
```

Comment:
Should we embrace nullability here?

```diff
    public MetricStreamIdentity(Instrument instrument, MetricStreamConfiguration? metricStreamConfiguration)
    {
-        this.MeterTags = instrument.Meter.Tags ?? Enumerable.Empty<KeyValuePair<string, object?>>();
+        this.MeterTags = instrument.Meter.Tags;
    }

-    public IEnumerable<KeyValuePair<string, object?>> MeterTags { get; }
+    public IEnumerable<KeyValuePair<string, object?>>? MeterTags { get; }
```

Thinking the way it is currently there isn't a super cheap way to skip processing for the common case where it isn't set.

```csharp
    AddAttributes(meterTags, metrics.Scope.Attributes); // pay for a call

    private static void AddAttributes(IEnumerable<KeyValuePair<string, object>> meterTags, RepeatedField<OtlpCommon.KeyValue> attributes)
    {
        foreach (var tag in meterTags) // Pay to call GetEnumerator/MoveNext/check result
        {
            if (OtlpKeyValueTransformer.Instance.TryTransformTag(tag, out var result))
            {
                attributes.Add(result);
            }
        }
    }
```

VS:

```csharp
    if (meterTags != null) // Pay for a null check only
    {
       AddAttributes(meterTags, metrics.Scope.Attributes);
    }
```